src/driverstation/driverstation_gui/include/driverstation_gui/dynamic_texture.hpp

```diff
@@ -0,0 +1,442 @@
+					}
+				}
+			}
+
+			util::rendering::FlipVertically(this->render_texture);
+
+			return DynamicRenderTexture2D::current_texture();
+		}
+
+		void on_click(float x, float y) override{
+			float x_in_cells = x * row_count;
+			float y_in_cells = y * column_count;
+
+			uint8_t inter_cell_x = std::floor(x_in_cells);
+			uint8_t inter_cell_y = std::floor(y_in_cells);
+
+			float intra_cell_x = fmod(x_in_cells, 1.0f);
+			float intra_cell_y = fmod(y_in_cells, 1.0f);
+
+			for(uint8_t y = 0; y <= inter_cell_y; y++){
+				for(uint8_t x = 0; x <= inter_cell_x; x++){
+					MultiCellDynamicTextureGrid2DCell* cell = &this->cells[y][x];
+					if(cell->value){
+						uint8_t end_x = x + cell->cell_count_x - 1;
+						uint8_t end_y = y + cell->cell_count_y - 1;
+
+						if(end_x >= inter_cell_x && end_y >= inter_cell_y){
+							float x_in_component = ((inter_cell_x - x) + intra_cell_x) / (cell->cell_count_x);
+							float y_in_component = ((inter_cell_y - y) + intra_cell_y) / (cell->cell_count_y);
+
+							cell->value->on_click(x_in_component, y_in_component);
+						}
+					}
+				}
+			}
+		}
+
+		MultiCellDynamicTextureGrid2DSelectFrom<row_count, column_count> from(uint8_t x, uint8_t y){
+			return MultiCellDynamicTextureGrid2DSelectFrom(this, x, y);
+		}
+
+		std::optional<std::shared_ptr<DynamicTexture2D>> get(uint8_t x, uint8_t y){
+			MultiCellDynamicTextureGrid2DCell* cell = &this->cells[y][x];
+			if(cell->value){
+				return cell->value;
+			}else{
+				return std::nullopt;
+			}
+		}
+
+		private:
+			// TODO: FIXME: MultiCellDynamicTextureGrid2DCell type is defined in anonymous namespace but is in header file
+			std::array<std::array<MultiCellDynamicTextureGrid2DCell, column_count>, row_count> cells;
+			Vector2 cell_view_size;
+	};
+
+	template<uint8_t row_count, uint8_t column_count>
+	struct MultiCellDynamicTextureGrid2DSelectFrom{
+		friend struct MultiCellDynamicTextureGrid2D<row_count, column_count>;
+
+		MultiCellDynamicTextureGrid2DSelectFromTo<row_count, column_count> to(uint8_t x, uint8_t y){
+			return MultiCellDynamicTextureGrid2DSelectFromTo(
+				this->multi_cell,
+				this->from_x, this->from_y,
+				x, y
+			);
+		}
+
+		private:
+			MultiCellDynamicTextureGrid2D<row_count, column_count>* multi_cell;
+			uint8_t from_x;
+			uint8_t from_y;
+
+			MultiCellDynamicTextureGrid2DSelectFrom(
+				MultiCellDynamicTextureGrid2D<row_count, column_count>* multi_cell,
+				uint8_t from_x, uint8_t from_y
+			): multi_cell(multi_cell), from_x(from_x), from_y(from_y){}
+	};
+
+	template<uint8_t row_count, uint8_t column_count>
+	struct MultiCellDynamicTextureGrid2DSelectFromTo{
+		friend struct MultiCellDynamicTextureGrid2DSelectFrom<row_count, column_count>;
+
+		MultiCellDynamicTextureGrid2D<row_count, column_count>* set(std::shared_ptr<DynamicTexture2D>&& value){
+			this->multi_cell->cells[this->from_y][this->from_x] = MultiCellDynamicTextureGrid2DCell{
+				std::move(value), this->to_x - this->from_x, this->to_y - this->from_y
```

Comment:
Compiler is mad about narrowing conversion but these are both uint8_ts. Surround in static_cast<uint8_t>() to make it shut up